pkg/plugin/clientmgmt/process.go

```diff
@@ -59,7 +61,51 @@ func newProcess(command string, logger logrus.FieldLogger, logLevel logrus.Level
 	// This launches the plugin process.
 	protocolClient, err := client.Client()
 	if err != nil {
-		return nil, err
+		if !strings.Contains(err.Error(), "unknown flag: --features") {
+			return nil, err
+		}
+
+		// Velero started passing the --features flag to plugins in v1.2, however existing plugins
+		// may not support that flag and may not silently ignore unknown flags. The plugin server
+		// code that we make available to plugin authors has since been updated to ignore unknown
+		// flags, but to avoid breaking plugins that haven't updated to that code and don't support
+		// the --features flag, we specifically handle not passing the flag if we can detect that
+		// it's not supported.
+
+		logger.Debug("Plugin process does not support --features flag, removing it and trying again")
+
+		var commandArgs []string
+		var featureFlag bool
+		for _, arg := range builder.commandArgs {
```

Comment:
Suggest pulling this loop into a separate function that we can have unit tests for.

Reply:
done!

Reply:
Thank you!